cudamat/cudamat.py

```diff
@@ -1501,6 +1503,25 @@ def where(condition_mat, if_mat, else_mat, target=None):
     return target
 
 
+def correlate(mat, kernel, target=None):
+    """
+    Cross-correlate a matrix with a kernel matrix.
+    The kernel matrix is centered over each element of the matrix mat.
+    Width and height of the kernel matrix must be an odd integer.
```

Comment:
When you're changing the `target = ` line, could you add two sentences here, please:
```
If a target is not provided, a new matrix is created for storing the result.
Note that this function cannot operate in-place.
```